lib/pure/collections/sequtils.nim

```diff
@@ -281,6 +281,19 @@ when (NimMajor, NimMinor) <= (1, 0):
 else:
   zipImpl(s1, s2, seq[(S, T)])
 
+proc unzip*[S, T](s: openArray[(S, T)]): (seq[S], seq[T]) =
+  ## Returns a tuple of two sequences split out from a sequence of 2-field tuples.
+  runnableExamples:
+    let
+      zipped = @[(1, 'a'), (2, 'b'), (3, 'c')]
+      unzipped1 = @[1, 2, 3]
+      unzipped2 = @['a', 'b', 'c']
+    assert zipped.unzip() == (unzipped1, unzipped2)
+    assert zip(unzipped1, unzipped2).unzip() == (unzipped1, unzipped2)
+  for elem in s:
```

Comment:
We should make sure to initialize the `result` sequences with the correct length, since it's the same as the length of `s`.
```suggestion
  result[0] = newSeqOfCap[S](s.len)
  result[1] = newSeqOfCap[T](s.len)
  for elem in s:
```